### ISSUE TITLE
continuous listening without wake word

### ISSUE THREAD

**Issue by surajhes** [alexa/alexa-avs-sample-app#797]
HI  I was wondering if its possible to have Alexa on Raspberry Pi and not have to wake her up with wake word everytime? Like just once wake it up and keep on listening till we explicitly kill it or shutdown it ?

**Comment by shaolinkhoa** [comment]
You can use  ["GPIO wake word agent"](https://github.com/alexa/alexa-avs-sample-app/issues/619#issuecomment-291957607) to wake alexa up by button. NOTE: only works on raspberry.

Or set `"shouldEndSession": false` in [response request](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/alexa-skills-kit-interface-reference#request-format).

**Comment by surajhes** [comment]
I was using flask-ask framework for sending a response back as statement, where the session ends after we respond as statement. (Making alexa read out a return speech) So after the response, it would still be awake and i dont have to explicitly say "Alexa" and wake it up?

**Comment by shaolinkhoa** [comment]
Statement will end the conversation so I use question at the return of each intent.
Currently, I'm  using [afg](https://github.com/dgtony/afg) to work with Flask-ask, a library for building scenario-based conversational skills. It's a state machine. 
I build 2 states: **Ask** and **Process**. After Process's intent was finished, it will jump back to state ask and wait for my next question. 
I really don't know how long Alexa open for a session in conversation.

**Comment by pierceds** [comment]
That's an interesting use-case! We don't support this option out-of-the-box on the Java sample app right now. A few callouts below:

- Internally Alexa Service cuts off a request after it has streamed a large chunk of data. This is to keep our traffic manageable, and would perhaps pose problems for this type of user experience at present.
- How would you envision handling directives sent back from Alexa Service for this experience? Would you stop recording, handle the directive, then immediately start recording again? What if you receive a [StopCapture](https://developer.amazon.com/public/solutions/alexa/alexa-voice-service/reference/speechrecognizer#stopcapture) directive?
- What is the expected user experience for this approach? Would they only be speaking with Alexa (i.e. not talking to other people in the same room)? I imagine the continuous streaming approach would lead to a lot of unusable audio being streamed if not. This could potentially lead to accuracy problems where Alexa responds incorrectly.
- As per the [AVS Terms of Service](https://developer.amazon.com/public/solutions/alexa/alexa-voice-service/support/terms-and-agreements), you need to ensure that end users receive a prominent notice of: 
  1) the fact that your device is streaming audio data to Alexa and 
  2) the ways to start/stop streaming using your device (how would you handle this?)

**Comment by shaolinkhoa** [comment]
Hi @pierceds 
As you said, there are a lot of problems to handle if we want to keep Alexa listening.
That why,for convenience purposes, I use "GPIO wake word agent" to wake alexa up. 

I think surajhes want Alexa keep listening and analyzing the conversation then Alexa will recommend a solution. Ex: you told your friend that you are hungry, then Alexa will automatically  recommend a place to eat
But it's belong to AI assistant category.

**Comment by pierceds** [comment]
Indeed, both "Push-to-talk" and "Tap-to-talk" are well supported Alexa use-cases today. 

@surajhes Let us know if the above answers your original question!
